File: 3567-convert-date-to-binary/convert-date-to-binary.cpp
class Solution {
public:
    string bin (int num) {
        string result = "";
        while (num > 0) {
            result = to_string(num % 2) + result;
            num = num / 2;
        }
        return result;
    }
    string convertDateToBinary(string date) {
        string y = date.substr(0, 4);
        string m = date.substr(5, 2);
        string d = date.substr(8, 2);
        return bin(stoi(y)) + "-" + bin(stoi(m)) + "-" + bin(stoi(d));
    }
};

File: 3567-convert-date-to-binary/convert-date-to-binary.java
class Solution {
    public String bin(int num) {
        StringBuilder s = new StringBuilder();
        while (num > 0) {
            s.append(String.valueOf(num % 2));
            num = num / 2;
        }
        s.reverse();
        return s.toString();
    }
    public String convertDateToBinary(String date) {
        int y = Integer.parseInt(date.substring(0, 4));
        int m = Integer.parseInt(date.substring(5, 7));
        int d = Integer.parseInt(date.substring(8));

        return bin(y) + "-" + bin(m) + "-" + bin(d);
    }
}